backend/question-service/lambda/app/scraper/handler.py

```diff
@@ -0,0 +1,189 @@
+import time
+import datetime
+import requests
+from bs4 import BeautifulSoup, XMLParsedAsHTMLWarning
+from pymongo import MongoClient, UpdateOne
+import json
+from pymongo.errors import ConnectionFailure, ServerSelectionTimeoutError, BulkWriteError
+from dotenv import load_dotenv
+import os 
+import warnings
+import concurrent.futures
+
+# Load environment variables
+load_dotenv()
+MONGO_CONNECTION_STRING = os.getenv("MONGO_CONNECTION_STRING")
+RATE_LIMIT = int(os.getenv("RATE_LIMIT", 60)) # Default to 60 questions per function invokation - the Rate Limit
+COOLDOWN_PERIOD = int(os.getenv("COOLDOWN_PERIOD", 10 * 24 * 60 * 60)) # Default to 10 days in seconds
+
+# Set MongoDB Client and Connection as a global variable
+client = None
+collection = None
+
+def handler(event, context):
+    """Lambda function handler. Scrapes Leetcode for questions and saves them to the MongoDB database."""
+
+    # TODO: Implement custom event handling logic
+    print("The event received was: ", event)
+
+    # Establish a connection to the MongoDB server
+    global client
+    client = connect_to_mongo(MONGO_CONNECTION_STRING)
+
+    # Select the database and collection
+    db = client['questions_db']
+    global collection
+    collection = db['problems']
+
+    all_questions = parse_questions_xml()
+    questions_to_scrape = get_questions_to_scrape(all_questions)
+    problems = scrape_leetcode_from_urls(questions_to_scrape)
+    save_to_mongodb(problems)
+
+    # Close the connection
+    client.close()
+
+    return f"Successfully scraped and saved {len(problems)} questions!"
+
+def parse_questions_xml():
+    """Parse the Leetcode questions sitemap XML file and return a list of all question URLs."""
+    BASE_URL = "https://leetcode.com/sitemap/sitemap-question-1.xml"
```

Comment:
nit: if it is not too difficult to separate into multiple python files/modules, can abstract out the constants into another file for readability?